lib/tx-core-models.ts

```diff
@@ -13,9 +13,10 @@ export class TxMessage {
   constructor(
     readonly msgIndex: number,
     readonly requestType: string,
-    readonly details: any,
   ) {
   }
+
+  details: any = { }; // details are always empty.
```

Comment:
I guess it is mutable, which means a client can do like below.
```
txMessage.details["new-property"] = "new property value";
```

So it'd better `readonly details: any = { };`